### ISSUE TITLE
Plugin Greyed Out Again But With a Different Error

### ISSUE THREAD

**Issue by cyanidearsenic** [Grum999/BuliBrushSwitch#18]
I am trying to upgrade the 1.0 release now that I've updated to Krita 5.2.1 and the plugin is (again) greyed out in the plugin manager.
![Screenshot_1](https://github.com/Grum999/BuliBrushSwitch/assets/12106286/1ab334e8-10ac-4eb3-9bb1-64378fea5879)
This is the error that is showing up on the mouse-over.
I wasn't sure about whether it was easy to upgrade in place, so I removed the plugin from the newly updated Krita and entirely reinstalled it.
This is the 1.0 BuliBrushSwitch, on Krita 5.2.1, on Windows 11.

**Comment by Grum999** [comment]
Hi

Thanks for report!

For what I see there's a bug when trying to retrieve gradients resources 🫤
I need to dig.

When you said you've removed the plugin, did you also removed the plugin configuration file?
Should be in your case `C:\Users\Alex\AppData\Local\krita-plugin-bulibrushswitchrc.json`


Not sure if it will change something, eventually, rename it (to not loose it) and restart krita

Anyway even if this fix the problem, there's something wrong in the code to review...

**Comment by cyanidearsenic** [comment]
I did not remove the configuration file before, I completely forgot there would be one.
I tried taking the plugin back out, moving the config file, and reinstalling, but it's the same error still

**Comment by Grum999** [comment]
Ok thanks for information.
I'll try to reproduce the case on my side.

I can fix the "crash" itself it's not difficult (add a new condition to avoid access to a null value and then avoid crash), but for now the condition why the value is not initialized is not clear to me (so, it could generate other problems...)

Another question: do you have some gradients defined in your resources?
The plugin crashes when trying to retrieve them...

**Comment by cyanidearsenic** [comment]
Yes, I think so?
I have a Rakurri Gradient Map set and a few things that come up when I look through the gradients section in 'Manage Resources'.
Are those what you meant?

**Comment by Grum999** [comment]
Yes that's it.
Thanks.

I'll install it too, not sure it's related but I need to look at all possible case.

**Comment by Grum999** [comment]
Hi @cyanidearsenic 

Ok, I have something 
- Everything is fine using Krita AppImage on Linux 
- I have the same problem than you using Krita on Windows 

So I'll be able to dig and search for a solution 🙂
Many thanks for your feedback!

**Comment by Grum999** [comment]
I've fixed bug on my side, need to do some additional tests and I'll try to publish release this week-end.

**Comment by Grum999** [comment]
Plugin v1.0.1 published
Cant you try it and tell me if it's Ok now?

**Comment by cyanidearsenic** [comment]
I've downloaded it, and it installed perfectly and seems to be working now without any errors.

**Comment by cyanidearsenic** [comment]
> I've downloaded it, and it installed perfectly and seems to be working now without any errors.

Oops, spoke too soon. When I attempted to add a brush back in and edit the settings for it, I get another error.
I'm not sure if it's the same issue or a new one, would you like the error here? It starts out with
```
A problem occurred in a Python script.  Here is the sequence of
function calls leading up to the error, in the order they occurred.

 C:\Users\Alex\AppData\Roaming\krita\pykrita\bulibrushswitch\bbs\bbsmainwindow.py in __actionEdit(self=<bulibrushswitch.bbs.bbsmainwindow.BBSMainWindow object>)
```

**Comment by Grum999** [comment]
Hi

Sorry for this error message, it seems something has been broken with v1.0.1
I can reproduce it, I've created issue #19 to track/fix it.

As it is an another error, I'll close this issue and fix problem from the #19

**Comment by Grum999** [comment]
@cyanidearsenic 
v1.0.2 published for issue #19 
Hope this is the right one 🤞
